filter_plugins/ec2_custom.py

```diff
@@ -130,12 +140,16 @@ def get_sg_id_result(result_list):
 
 def get_launch_configs(launch_configs, stack_name):
     import json
+    import re
+
     configs = []
     launch_configs = json.loads(launch_configs)
 
     if 'LaunchConfigurations' in launch_configs:
+        component_regex = "({})".format('|'.join(VALID_APP_COMPONENT_NAMES))
         for launch_config in launch_configs['LaunchConfigurations']:
-            if stack_name in launch_config['LaunchConfigurationName']:
+            pattern = re.compile("^{}-{}$".format(component_regex, stack_name))
+            if pattern.match(launch_config['LaunchConfigurationName']):
```

Comment:
{code}
14:55:51 TASK [ec2-app : Extract stack specific lc's] ***********************************
14:55:51 ok: [localhost] => {"ansible_facts": {"env_launch_configs": []}, "changed": false}
14:55:51 
{code}

It hasn't selected any configs to delete